pkg/cdi/device.go

```diff
@@ -0,0 +1,74 @@
+/*
+   Copyright © 2021 The CDI Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package cdi
+
+import (
+	cdi "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+// Device represents a CDI device of a Spec.
+type Device struct {
```

Comment:
I don't quite follow why we need to maintain the link between the `cdi.Device` and the `Spec` -- especially since we don't perform any validation on the `spec` member.

Reply:
We'd want to support things like being able to ask/tell about a looked up *Device, without extra information, what is its fully qualified name, vendor, class, etc. which all is a bunch of information associated with the Spec the device was loaded from. To do this you basically have two choices: you either need to know which Spec the device was loaded from or you need to duplicate this information for the Device. I find keeping the association a better choice.